malariagen_data/ag3.py

```diff
@@ -2489,6 +2490,89 @@ def aa_allele_frequencies(
 
         return df_aaf
 
+    def plot_frequencies_heatmap(
+        self, df, index, max_len=100, y_label=None, colorbar=True, width=None
+    ):
+
+        """Plot a heatmap from a pandas DataFrame of frequencies, e.g., output from
+            `Ag3.snp_allele_frequencies()` or `Ag3.gene_cnv_frequencies()`. It's recommended to
+            filter the input DataFrame to just rows of interest, i.e., fewer rows than `max_len`.
+
+        Parameters
+        ----------
+        df : pandas DataFrame
+           A DataFrame of frequencies, e.g., output from `snp_allele_frequencies()` or `gene_cnv_frequencies()`.
+        index : str or list of str
```

Comment:
To make it even easier for the user, you could potentially have the option to specify if the table is either for SNP frequencies or CNVs and decide the indexes based on this information or perhaps update the doc string with details of the standard input.

Reply:
Hi @KellyLBennett, this is a good thought and we've had a bit of debate around this. I wonder if an even simpler approach would be to have the outputs the functions that create frequencies already set an index. E.g., the output of `snp_allele_frequencies()` is a dataframe with the appropriate index already set (contig, position, ref, alt). E.g., the output of `gene_cnv_frequencies()` is a dataframe with index gene_id already set. Then the `index` parameter here to `plot_frequencies_heatmap()` could be optional, and if not given would just use whatever index was already set. If so, the user could then do something like:

```python
df_freq = ag3.snp_allele_frequencies(...)
df_freq = df_freq.query(...)  # do some filtering
ag3.plot_frequencies_heatmap(df_freq)
```

Reply:
Thanks @KellyLBennett, @alimanfoo, I will have a look and see how that might work.